exonum/src/runtime/migrations.rs

```diff
@@ -0,0 +1,485 @@
+// Copyright 2019 The Exonum Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Migration tools.
+//!
+//! FIXME: more details
+
+// FIXME: Decouple after #1639 is merged.
+pub use semver::Version;
+
+use exonum_merkledb::migration::MigrationHelper;
+
+use std::{collections::BTreeMap, fmt};
+
+use crate::runtime::InstanceSpec;
+
+/// Atomic migration script.
+pub struct MigrationScript {
+    end_version: Version,
+    name: String,
+    logic: Box<dyn FnOnce(&mut MigrationContext) + Send>,
```

Comment:
Shouldn't it be `-> Result<(), ExecutionError>` (or `MigrationError`, dunno).

Reply:
Would you mind if I change it in a follow-up PR corresponding to ECR-4080?

Reply:
Sure!